### PR TITLE
(profiling-ffi) Expose telemetry through profiling ffi

### PR DESCRIPTION
**Pull request by paullegranddc**
# What does this PR do?

* Expose telemetry-ffi from profiling-ffi
* Add a feature flag to disable it if necessary
* Expand telemetry builder macros to generate declarations from them with cbindgen
* Add a telemetry C example

# Motivation

# Additional Notes

Anything else we should know when reviewing?

# How to test the change?

Describe here in detail how the change can be validated.

## For Reviewers
- [ ] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

### CODE REVIEW

**Comment by gleocadie** [comment]
Can you update the [windows/build-artifacts.ps1](https://github.com/DataDog/libdatadog/blob/main/windows/build-artifacts.ps1) please ?
If you do not have any setup, I can do it for you

File: examples/ffi/telemetry.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present
+// Datadog, Inc.
+
```
Comment (r1viollet) [inline examples/ffi/telemetry.c:5]: thanks for re-adding the test!

File: ddtelemetry-ffi/src/builder/expanded.rs [hunk 2]
```diff
@@ -0,0 +1,1043 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+
+pub use macros::*;
```
[the remark this answers is not in the document]
Comment (paullegranddc) [reply ddtelemetry-ffi/src/builder/expanded.rs:4]: I think it's unlikely.
The expanded code is gated by a feature, which is enabled by default, and when the cbindgen feature is.
So if you re-generate header, the shared library should contain the code from expanded.rs so any macro modification would be ignored.

The only way to have miss-matches would be to generate headers in one step, and then to generate the artifact disabling default features in a second one which doesn't make a lot of sense

**Review by r1viollet — APPROVED** [review]
LGTM
We will have a bit of a clash with my symbolization PR, but nothing too important.
We should just talk on what make sense as default build, and what should be added optionally (in build profiling / other specific builds).

**Pull request merged**